### PR TITLE
[VL] Apply patch to Velox for compilation

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Removes build script changes to Velox by applying patch from Gluten to Velox.
https://github.com/oap-project/velox/pull/425

## How was this patch tested?

under test

### CODE REVIEW

File: ep/build-velox/src/build_velox.sh [hunk 1]
```diff
@@ -65,6 +65,16 @@ for arg in "$@"; do
...
 done
 
+function apply_compilation_fixes {
+  current_dir=$1
+  velox_home=$2
+  cp ${current_dir}/compilation.patch ${velox_home}/
+  sudo mkdir -p ${velox_home}/third_party/arrow_patches/
+  sudo cp ${current_dir}/helpers.patch ${velox_home}/third_party/arrow_patches/
+  cd ${velox_home}
+  git apply compilation.patch
```
Comment (PHILO-HE) [inline ep/build-velox/src/build_velox.sh:75]: If `git apply` fails, will the build script fail or continue the build even though the patch is not applied? Could you confirm it? Thanks!

Comment (rui-mo) [reply ep/build-velox/src/build_velox.sh:75]: If it fails, I suppose whether to continue or to stop, the build process will fail anyway. How do you think?

Comment (zhouyuan) [reply ep/build-velox/src/build_velox.sh:75]: not actually, as it may build the original code?

Comment (PHILO-HE) [reply ep/build-velox/src/build_velox.sh:75]: I think we can do a check after executing `git apply`. And leave some message to help developers easily figure out why build failed.

Reference: https://askubuntu.com/questions/29370/how-to-check-if-a-command-succeeded

Comment (rui-mo) [reply ep/build-velox/src/build_velox.sh:75]: Updated. Thanks.

File: ep/build-velox/src/build_velox.sh [hunk 2]
```diff
@@ -65,6 +65,19 @@ for arg in "$@"; do
...
+function apply_compilation_fixes {
+  current_dir=$1
+  velox_home=$2
+  sudo cp ${current_dir}/compilation.patch ${velox_home}/
+  sudo mkdir -p ${velox_home}/third_party/arrow_patches/
+  sudo cp ${current_dir}/helpers.patch ${velox_home}/third_party/arrow_patches/
+  cd ${velox_home}
+  git apply compilation.patch
+  if [ $? -ne 0 ]; then
+    echo "Failed to apply compilation fixes to Velox: $?."
```
Comment (PHILO-HE) [inline ep/build-velox/src/build_velox.sh:77]: Please add `exit 1` to directly make the build exit. Suppose it is not acceptable to continue the build without the patch applied.

**Review by PHILO-HE — COMMENTED** [review]
Nit:
Rename `compilation.patch` to some name like `modify_velox_build.patch`, which can be more read-friendly to developers.

File: ep/build-velox/src/build_velox.sh [hunk 3]
```diff
@@ -65,6 +65,19 @@ for arg in "$@"; do
   esac
 done
 
+function apply_compilation_fixes {
+  current_dir=$1
+  velox_home=$2
+  sudo cp ${current_dir}/modify_velox.patch ${velox_home}/
+  sudo cp ${current_dir}/modify_arrow.patch ${velox_home}/third_party/
```
Comment (JkSelf) [inline ep/build-velox/src/build_velox.sh:72]: @rui-mo Do we need to git apply modify_arrow.patch like modify_velox.patch in  line 74?

File: ep/build-velox/src/build_velox.sh (hunk 2 above)
Comment (rui-mo) [reply ep/build-velox/src/build_velox.sh:76]: Updated.

File: ep/build-velox/src/build_velox.sh (hunk 3 above)
Comment (rui-mo) [reply ep/build-velox/src/build_velox.sh:72]: It is applied to Arrow during compilation: https://github.com/oap-project/gluten/pull/3520/files#diff-b41402fae332ead78c4646843c9bcf66a2813203cdc82521c9c9f130b48146dbR100.

**Review by PHILO-HE — COMMENTED** [review]
LGTM!

**Review by PHILO-HE — APPROVED** [review]

**Pull request merged**